### DISCUSSION TITLE
My country does not appear in the options

### DISCUSSION THREAD

**Discussion by martinhash** [community/community#22814] [New to GitHub]
<p>Hello, im from Chile and i try to add my celphone to Autenticator, and not appear in the options, somebody can help me?</p>

**Comment by Sanjeev9910** [comment]
<p>Maybe you entering the wrong Pincode. Check it once. If not work then enter the Pincode your surrounding city.</p>

**Comment by canuckjacq** [comment, marked as answer]
<p>:wave: Welcome!</p>
<p>I’m afraid our SMS providers can’t ensure reliability of service in all countries, so there are definitely a few missing from the list. We add new countries whenever we can. In the meantime, you are best using a TOTP application (anything like Google Authenticator or Duo) for 2FA.</p>

**Reply by PeterDualCore** [reply]
we can expect Brazil to enter in the sms list?

**Reply by lingfeng01** [reply]
What about China?

**Reply by KateTheJourneyer** [reply]
Same with Thailand.

**Reply by abdu218x** [reply]
same with egypt

**Reply by ajuliush** [reply]
Same with Bangladesh

**Reply by dsifat** [reply]
need Bangladesh +880

**Reply by Mourad-Tlili** [reply]
Same as Tunisia +216

**Reply by AshirButt925** [reply]
Same as Pakistan ```+92```

**Reply by Crymfox** [reply]
Same as Tunisia +216

**Reply by Technaid-void** [reply]
Same as Lebanon +961

**Reply by shadowofaroman** [reply]
Same as Kenya.

**Reply by evanalsadik** [reply]
same as indonesia

**Comment by pabloazurduy** [comment]
Hello, I was wondering if it's possible to add Chile to the list of countries where SMS is available?. Many apps have the SMS available (google, meta*, etc), I actually can't recall any app that the SMS service is not working in chile besides Github. Any chance that we push to add it to the list?

Thanks!

**Reply by joshuawalker** [reply]
Hey @pabloazurduy—we're always reviewing our list of supported countries for SMS authentication. In the meantime, we'd recommend using an authenticator application (e.g. Microsoft Authenticator, 1Password) which is often a more secure authentication method that doesn't depend on message deliverability.

**Reply by pabloazurduy** [reply]
hey @joshuawalker thanks for the reply, yes I do use 2FA with the (GitHub) app for a long time now. Still, I would like to have another recovery system just because when you have a phone lost, there is a lot of 2FA that keeps you out of your email/account, and the phone number is something you can recover (at least in Chile). 
On a daily basis, I'm ok just using the app/duo, but when you need to recover an account the SMS system is a critical feature.

**Reply by kheitmannf** [reply]
Hey @joshuawalker , how can it be possible that GITHUB, the website that has sent to the arctic vault software to be buried 250 meters deep under the permafrost to be preserved for future generations, does not have the complete list of countries in the SMS 2FA alternative?! People say that TOTP is more secure. But if I lost my phone or somebody stole it I'll get locked out of my account!!! I'm a github user since 2012, and have more than 150 public repositories (and many many many more privates), do you know the amount of work that I would lose?

Have you ever tested your recovery codes? I tested them by configuring TOTP in another smartphone... and it does not work!!! it has bugs!!!! If you put both phones side by side the TOTP codes don't match. Do the TOTP output depends on the language (locale) the phone is configured? If you have a cellphone in english, and another cellphone in spanish the TOTP codes won't match? How can I trust such a system? SMS can sometimes be unavailable, or sometimes you will have to wait 10 minutes, or 1 hour. But at least you will GET your SMS code. It is a robust system that works. But my experience with TOTP is that it is not reliable. But who cares? Who tests software these days? Everybody says testing is boring and not fun. Everybody trusts that they cellphone will never get lost or will never be stolen. Everybody trusts that TOTP is realiable.

**Reply by kheitmannf** [reply]
@pabloazurduy  La razón no es que tienen una lista de países que "soportan" SMS. Porque si Chile no soportara SMS, tampoco estaría en la lista de Microsoft, Meta, etc. La razón es porque estos vagos copiaron la lista en algun momento y después nunca más la volvieron a actualizar: total tenemos TOTP authenticator y nosotros creemos que la gente tiene que usar TOTP, entonces no vamos a actualizar la lista de SMS y a los wnes que pregunten les decimos que "estamos trabajando en ello". Así son los progres (que están por todas partes en github): los wnes más dictadores del mundo que quieren que todos les hagamos caso en sus ideas. Por eso que el software cada vez funciona más como el loli wn.

**Comment by Neodevils** [comment]
Okay, that's good to know @canuckjacq. We can also just deploy our Secret keys on any cloud services too @pabloazurduy. I know that's sound weird but it's the best way until GitHub is available to send messages to all countries. I'm from Türkiye and I can't add SMS fallback either...

**Reply by pabloazurduy** [reply]
Hi, what do you mean? for us to deploy a service that gives us an SMS authentication?

**Reply by Neodevils** [reply]
> Hi, what do you mean? for us to deploy a service that gives us an SMS authentication?

I was meant, you can save the back-up codes on somewhere else. If there's an event happening about your phone, you can just restore your GitHub or any app uses 2FA system in that way. 👍🏻

**Reply by pabloazurduy** [reply]
ohh I do, (probably not a good practice given that I might lose access to that place too or someone else get them, but anyway). thks !

**Reply by Neodevils** [reply]
> ohh I do, (probably not a good practice given that I might lose access to that place too or someone else get them, but anyway). thks !

You're welcome.

**Comment by vahekhumaryan** [comment]
Can you please add Armenia to the list.

**Comment by vitobasso** [comment]
Same for Brazil :~
As mentioned by [pabloazurduy](https://github.com/pabloazurduy) github is the only website I use that doesn't include these countries, I have no problem with google, facebook, wise, hotmail, etc.

**Reply by Hadaward** [reply]
Yea 😥i prefer to use SMS, it's just weird that Brazil is not in the list, every app i know let you use any number you want.

**Comment by jughoor** [comment]
Pakistan is also missing.

**Reply by deansmith786** [reply]
Really

**Comment by erik-ingwersen-ey** [comment]
Hey GitHub Team,

Just wondering if there's any update on adding more countries to the 2FA SMS support list, especially places like Brazil, Chile, and Pakistan. It's great that there are other ways to do 2FA, but SMS would really help a bunch of us out.

About the whole 'SMS providers aren’t reliable in all countries' thing - I'm not sure that really flies, especially for big economies like the ones mentioned. It feels like GitHub might not be super keen on pushing out SMS 2FA more broadly. If that's the case, it’d be cool to just say it straight up instead of the provider reliability argument.

Would love to get some clarity on this. **Is expanding SMS 2FA something on GitHub's radar, or should we just not get our hopes up?** Thanks for any info you can share!

**Reply by Futuza** [reply]
I'm guessing, that its now 2025, and nothing has been answered and these countries SMS 2FA remain blocked/unavailable, Github/Microsoft does not intend to help with this.

**Comment by raza2022** [comment]
Yesterday I received the message 
`Your account meets this criteria, and you will need to enroll in 2FA within 45 days, by January 26th, 2024 at 00:00 (UTC). After this date, your access to GitHub.com will be limited until you enroll in 2FA. Enrolling is easy, and we support several options, starting with TOTP apps and text messages (SMS) and then adding on passkeys and the GitHub Mobile app.`

When I tried to enable SMS/Text messages, I was surprised to see that `PAKISTAN+92` was not in the list, I researched more and saw many countries missing from that list, how on earth can an enterprise do this to push your time users without complete mandatory steps, 
There should be an `official response/message from GitHub,` because as far as I know my country is in 4th position in freelancer/tech-based economies
I do not wanna seem bitter, but sadly, now I remember my friends who suggested moving from GitHub after the Microsoft acquisition

**Comment by time364** [comment]
Pakistan does not appear in the list.

**Comment by doooooo** [comment]
Also my country Egypt not in the list ! .. i'm not able to authenticate using TOTP, as the code generated is always invalid !

**Comment by ch3st3r08** [comment]
My Country El Salvador is not in the last, and it'd be awesome if you guys can put it in the list.

**Comment by SkandarS0** [comment]
Tunisia doesn't exist

**Comment by jorie100** [comment]
Chile is not there, and I also want to know is there and specific reason of why on of the biggest countries in the world, China, is not there also? I've seen plenty of pages with both codes.

**Comment by CrisA-Works** [comment]
What about Argentina?

**Comment by anasalaqeel** [comment]
I'm Saudi and I moved to bitbucket, they support Saudi Arabia phones and they have better user experience, better UI, more flexibility, better support. Bye github

**Comment by YOHANNES-DEGEFA** [comment]
Ethiopia is also missing

**Comment by polin103109** [comment]
Bangladesh is missing in the list.

**Comment by nicoaviguey** [comment]
Plenary 2025 and they still don't have Chile on the list. I've never used a service that didn't have it before; having to use an external app isn't convenient. :( At least I don't use external apps for 2FA at all. Now I'll have to download one just from GitHub. I hope they fix this soon.

**Comment by Crymfox** [comment]
Tunisia isn't in the list as well, please consider adding the missing countries asap x))

**Comment by kopo6ap** [comment]
+389 Macedonia not on the list
